tests/test_datasets.py

```diff
@@ -1,62 +1,82 @@
 import base64
-import re
 import json
+import tempfile
 
 from click.testing import CliRunner
 import responses
 
 from mapboxcli.scripts.cli import main_group
+from mapboxcli.scripts import datasets
+from mapboxcli.errors import MapboxCLIException
+
 import mapbox
 
 username = 'testuser'
 access_token = 'sk.{0}.test'.format(
     base64.b64encode(b'{"u":"testuser"}').decode('utf-8'))
+service = mapbox.Datasets(access_token=access_token)
+
+null_island = {
+    "type": "FeatureCollection",
+    "features": [
+        {
+            "id": "a",
+            "type": "Feature",
+            "properties": {"name": "Null"},
+            "geometry": {"type": "Point", "coordinates": [0, 0]}
+        }
+    ]
+}
+
 
 @responses.activate
-def test_cli_dataset_list_stdout():
-    datasets = """
-        [
-         {{
-          "owner":"{username}",
-          "id":"120c1c5aec87030449dfe2dff4e2a7c8",
-          "name":"first",
-          "description":"the first one",
-          "created":"2015-07-03T00:14:09.622Z",
-          "modified":"2015-07-03T00:14:09.622Z"
-         }},
-         {{
-          "owner":"{username}",
-          "name":"second",
-          "description":"the second one"
-          "id":"18571b87d4c139b6d10911d13cb0561f",
-          "created":"2015-05-05T22:43:10.832Z",
-          "modified":"2015-05-05T22:43:10.832Z"
-         }}
-        ]""".format(username=username)
+def test_cli_datasets_create_tileset():
+    dataset = "dataset-1"
+    name = "test"
+    tileset = "testuser.data"
+    owner = tileset.split('.')[0]
 
-    datasets = "".join(datasets.split())
+    expected = """{{
+        "id":"ciiae5gc40041mblzuluvu7jr",
+        "name":"{0}",
+        "complete":false,
+        "error":null,
+        "created":"2015-12-17T15:17:46.703Z",
+        "modified":"2015-12-17T15:17:46.703Z",
+        "tileset":"{1}",
+        "owner":"{2}",
+        "progress":0
+    }}""".format(name, tileset, owner)
 
     responses.add(
-        responses.GET,
-        'https://api.mapbox.com/datasets/v1/{0}?access_token={1}'.format(username, access_token),
+        responses.POST,
+        'https://api.mapbox.com/uploads/v1/{0}?access_token={1}'.format(
+            owner, access_token
+        ),
+        status=201, body=expected,
         match_querystring=True,
-        body=datasets, status=200,
-        content_type='application/json')
+        content_type='application/json'
+    )
 
     runner = CliRunner()
     result = runner.invoke(
         main_group,
         ['--access-token', access_token,
          'datasets',
-         'list'])
+         'create-tileset',
+         dataset,
+         tileset,
+         '--name', name])
 
     assert result.exit_code == 0
-    assert result.output.strip() == datasets.strip()
+    body = json.loads(responses.calls[0].request.body)
```

Comment:
@rclark here's the root of the Python 3 failure: `.body` is type `bytes`. We want `.text` which is type `str`. I think these assertions on the mock response objects can be eliminated anyway, this stuff is already tested by Responses. All we need to assert is `result.exit_code` and `result.output`.